core/node/libp2p/addrs.go

```diff
@@ -107,3 +130,41 @@ func ListenOn(addresses []string) interface{} {
 		}
 	}
 }
+
+func P2PForgeCertMgr() (*p2pforge.P2PForgeCertMgr, error) {
+	storagePath, err := config.Path("", "p2p-forge-certs")
+	if err != nil {
+		return nil, err
+	}
+
+	const authEnvVar = "FORGE_ACCESS_TOKEN"
+	const authForgeHeader = "Forge-Authorization"
+	authKey, foundAuthKey := os.LookupEnv(authEnvVar)
+
+	certMgr, err := p2pforge.NewP2PForgeCertMgr(
+		p2pforge.WithModifiedForgeRequest(func(req *http.Request) error {
+			if foundAuthKey {
+				req.Header.Set(authForgeHeader, authKey)
+			}
+			return nil
+		}),
```

Comment:
TODO: move to `p2p-forge/client` library, and call more specific funcs here:
- `WithForgeAuth(authKey)`
- `WithUserAgent(GetUserAgentVersion())`